samcli/commands/check/resources/Pricing.py

```diff
@@ -0,0 +1,155 @@
+        )
+        lambda_funciton_pricing.number_of_requests = user_input_requests
+
+        user_input_duration = _ask(
+            "What is the expected average duration of all lambda functions (ms)?", 1, self._max_duration
+        )
+        lambda_funciton_pricing.average_duration = user_input_duration
+
+        user_input_memory, user_input_unit = self._ask_memory(
+            'Enter the amount of memory allocated (128MB - 10GB), followed by a ":", '
+            "followed by a valid unit of measurement [MB|GB]"
+        )
+        lambda_funciton_pricing.allocated_memory = user_input_memory
+        lambda_funciton_pricing.allocated_memory_unit = user_input_unit
+
+        self._graph.lambda_function_pricing_info = lambda_funciton_pricing
+
+    def _correct_memory_input(self, user_input_split: List) -> bool:
+        """Checks if user input correct memory amount and unit
+
+        Args:
+            user_input_split (List): User enterd data [memory_amount, memory_unit]
+
+        Returns:
+            [type]: [description]
+        """
+        if len(user_input_split) != 2:
+            click.echo("Please enter a valid input.")
+            return False
+
+        memory_amount = user_input_split[0]
+        memory_unit = user_input_split[1]
+        valid_units = ["MB", "GB"]
+
+        try:
+            float(memory_amount)
+        except ValueError:  # Not a valid number
+            click.echo("Please enter a valid amount of memory.")
+            return False
+
+        if memory_unit not in valid_units:
+            click.echo("Please enter a valid memory unit.")
+            return False
+
+        # At this point, memory_amount and memory_unit are both valid inputs. Now test if memory_amount is within range
+
+        memory_amount_float = float(memory_amount)
+
+        if memory_unit == "GB":
+            # Convert to MB for testing range
+            memory_amount_float *= 1000
+
+        if (
+            memory_amount_float < self._min_memory_amount or memory_amount_float > self._max_memory_amount
+        ):  # units are in MB
+            click.echo("Please enter a valid amount of memory within the range.")
+            return False
+
+        return True
+
+    def _ask_memory(self, question: str) -> Tuple[float, str]:
+        """Ask user memory pricing question for lambda functions
+
+        Args:
+            question (str): The question to ask the user
+
+        Returns:
+            str: memory amount and unit
+        """
+        valid_user_input = False
+        user_input_split = []
+        while not valid_user_input:
+            user_input = click.prompt(text=question, type=str)
+            user_input_split = user_input.split(":")
+            if self._correct_memory_input(user_input_split):
+                valid_user_input = True
+
+        return user_input_split[0], user_input_split[1]
+
+
+def _ask(question: str, min_val: int = 1, max_val: float = float("inf")) -> int:
```

Comment:
this method is a utility method, I think it should be moved to a common module. I also can find the same method used here https://github.com/aws/aws-sam-cli/blob/3a5322c3cece6b79434b155fc38e61fd8e700d2a/samcli/commands/check/bottle_necks.py#L86